src/main/java/io/fabric8/elasticsearch/cloud/kubernetes/KubernetesAPIServiceImpl.java

```diff
@@ -28,20 +32,29 @@
 
   private final String namespace;
   private final String serviceName;
+  private final String podLabel;
 
   @Override
   public Endpoints endpoints() {
     logger.debug("get endpoints for service {}, namespace {}", serviceName, namespace);
     return client().endpoints().inNamespace(namespace).withName(serviceName).get();
   }
 
+  @Override
+  public List<Pod> pods() {
+    logger.debug("get endpoints with pod label {}, namespace {}", podLabel, namespace);
+    final String[] l = podLabel.split("=");
+    return client().pods().inNamespace(namespace).withLabel(l[0], l[1]).list().getItems();
```

Comment:
Some concerns I have here is whether it is possible to return a podlabel that is not in the pattern of 'foo=bar'?  It is possible to query for pods with labels that have a key but not a specific value.  Is the case we ALWAYS get key and value and will always be able to split it into 2 parts?  The value comes from settings?  Do we need a init check there to exit early if it is not in the expected format?